### DISCUSSION TITLE
classes/class_cameraserver

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#227] [User-contributed notes]
# classes/class_cameraserver

Inherits: Object Server keeping track of different cameras accessible in Godot. Description: The CameraServer keeps track of different cameras accessible in Godot. These are external cameras such a...

https://docs.godotengine.org/en/stable/classes/class_cameraserver.html

**Comment by codebykoi** [comment]
for the love of god, make it work on other Platforms

**Reply by Calinou** [reply]
CameraServer will be supported on Linux in 4.4: https://github.com/godotengine/godot/pull/53666

[It's being implemented on Windows too](https://github.com/godotengine/godot/pull/49763), but there is no ETA as the [4.x version of that pull request isn't in a mergeable state](https://github.com/godotengine/godot/pull/49763#issuecomment-1904592256).

**Reply by Calinou** [reply]
Camera support was added to Android in 4.5.dev:

- https://github.com/godotengine/godot/pull/106094

There's a pull request adding it to iOS and Web, but they haven't been merged yet as of writing:

- https://github.com/godotengine/godot/pull/106305
- https://github.com/godotengine/godot/pull/106784

**Comment by BubbleDoppel** [comment]
Hi! any ETA for support on Android ?
Thank you

**Reply by Calinou** [reply]
There's a pull request adding camera support on Android, but it needs to be split into several smaller pull requests before it can be merged (as it contains unrelated features).

- https://github.com/godotengine/godot/pull/98416

**Reply by Calinou** [reply]
Camera support was added to Android in 4.5.dev:

- https://github.com/godotengine/godot/pull/106094

**Comment by ronniebrito** [comment]
hi, 

I cant make webcam work using this class (ubuntu 22 godot 4.4) :/
could anyone provide a simple GDScript example to capture webcam image and set as a sprite texture?

thanks!!!

**Reply by Calinou** [reply]
Is the webcam recognized by other applications on the system (e.g. the web browser)? If so, please open an issue on https://github.com/godotengine/godot/issues with a minimal reproduction project attached.

**Comment by nickthunder007** [comment]
Hii, 
I want to access live camera stream for a simple AR project, however i cant get the stream. I'm a totle beginner so may sound stupid. 
I am using godot 4.4.1 with Ubuntu 22.04 and when I set feed.set_active(true) and check it again with feed.is_active() It is false. I am using meshinstance3D to display the feed but all it shows is pink checkerboard pattern in in-game camera view.
My pc has MJPEG and YUYV format if that helps. 

Any help is appreciated, Thank you in advance

**Reply by Calinou** [reply]
Linux camera support got some fixes in [4.5.beta1](https://godotengine.org/article/dev-snapshot-godot-4-5-beta-1/), give it a try: https://github.com/godotengine/godot/pull/105734

**Comment by anonomity** [comment]
Please this is the tilemap tutorial girl, please make it work on windows PLEASE

**Reply by Calinou** [reply]
This is the current PR adding Windows support:

- https://github.com/godotengine/godot/pull/108538

4.5 is in feature freeze, so any new features can only be merged in 4.6 at the earliest.
